src/transformers/modeling_tf_distilbert.py

```diff
@@ -663,15 +679,35 @@ def call(self, inputs, **kwargs):
             Attentions weights after the attention softmax, used to compute the weighted average in the self-attention
             heads.
         """
-        distilbert_output = self.distilbert(inputs, **kwargs)
+        if isinstance(inputs, (tuple, list)):
+            labels = inputs[6] if len(inputs) > 6 else labels
+            if len(inputs) > 6:
+                inputs = inputs[:6]
+        elif isinstance(inputs, (dict, BatchEncoding)):
+            labels = inputs.pop("labels", labels)
+
+        distilbert_output = self.distilbert(
+            inputs,
+            attention_mask=attention_mask,
+            head_mask=head_mask,
+            inputs_embeds=inputs_embeds,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            training=training,
+        )
 
         hidden_states = distilbert_output[0]  # (bs, seq_length, dim)
         prediction_logits = self.vocab_transform(hidden_states)  # (bs, seq_length, dim)
         prediction_logits = self.act(prediction_logits)  # (bs, seq_length, dim)
         prediction_logits = self.vocab_layer_norm(prediction_logits)  # (bs, seq_length, dim)
-        prediction_logits = self.vocab_projector(prediction_logits)
+        prediction_logits = self.vocab_projector(prediction_logits, training=training)
```

Comment:
@LysandreJik @VictorSanh - think this `training=training` was missing here when comparing to tf_bert. Not 100p sure though.

Reply:
I don't think that's necessary, as `self.vocab_projector` is a linear layer. I believe the `training` parameter is only useful for dropout?

Reply:
True! `training=training` is only relevant if there is a dropout or batchnorm keras layer: https://www.google.com/search?q=when+to+pass+training+parameter+keras&oq=when+to+pass+training+parameter+keras&aqs=chrome..69i57j69i64.4137j0j7&sourceid=chrome&ie=UTF-8